### PR TITLE
Feature filter lot at peremption risk

### LINKED ISSUE #5054: Improved filtering of data in the in-stock batch register
The batch register in stock allows data to be filtered according to several criteria except for:
- Product at risk of expiry
- As well as those with quantities at risk

image

### PR DESCRIPTION
**Pull request by lomamech**
- Addition of filtering of articles with expiration risks
-implementation of the test to trace batches with no risk of expiry and
without risk

closes #5054

### CODE REVIEW

File: server/controllers/stock/core.js [hunk 1]
```diff
@@ -301,6 +301,18 @@ async function getLotsDepot(depotUuid, params, finalClause) {
...
+  // Since the status of a product risking expiry is only defined
+  // after the comparison with the CMM, reason why the filtering
+  // is not carried out with an SQL request
+  if (params.is_expiry_risk === '1') {
+    return inventoriesWithLotsProcessed.filter(item => (item.S_RISK < 0 && item.lifetime > 0));
+  }
+
+  if (params.is_expiry_risk === '0') {
+    return inventoriesWithLotsProcessed.filter(item => (item.S_RISK >= 0 && item.lifetime > 0));
+  }
```
Comment (mbayopanda) [inline server/controllers/stock/core.js:314]: 👍 well done

**Review by mbayopanda — APPROVED** [review]
LGTM 👍

**Comment by lomamech** [comment]
All checks have passed

**Review by mbayopanda — APPROVED** [review]
👍

**Review by jniles — APPROVED** [review]
This has @mbayopanda's approval and it also looks good to me.

bors r+

**Pull request merged**